src/_image_resample.h

```diff
@@ -882,29 +700,34 @@ static void get_filter(const resample_params_t &params,
 }
 
 
-template<class T>
+template<typename color_type>
 void resample(
-    const T *input, int in_width, int in_height,
-    T *output, int out_width, int out_height,
+    const void *input, int in_width, int in_height,
+    void *output, int out_width, int out_height,
```

Comment:
This is so that the big switch below (`(type == NPY_UINT8) ? resample<agg::gray8> : ...`) works: as things are written currently, all template instantiations of resample have the same (function pointer) type `(*void)(const void*, int, ...)`.  If I explicitly put color_type instead, then they would have different types (the first argument would have different types) and the ternary would not work.